packages/near-membrane-dom/src/window.ts

```diff
@@ -178,65 +140,38 @@ function aggregateWindowDescriptors(
     return to;
 }
 
-/**
- * WindowProperties.prototype is magical, it provide access to any
- * object that "clobbers" the WindowProxy instance for easy access. E.g.:
- *
- *     var object = document.createElement('object');
- *     object.id = 'foo';
- *     document.body.appendChild(object);
- *
- * The result of this code is that `window.foo` points to the HTMLObjectElement
- * instance, and in some browsers, those are not even reported as descriptors
- * installed on WindowProperties.prototype, but they are instead magically
- * resolved when accessing `foo` from `window`.
- *
- * This forces us to avoid using the descriptors from the blue window directly,
- * and instead, we need to shadow only the descriptors installed in the brand
- * new iframe, that way any magical entry from the blue window will not be
- * installed on the iframe.
- */
-function aggregateWindowPropertiesDescriptors(
-    redOwnKeys: PropertyKey[],
-    blueDescriptors: PropertyDescriptorMap
-): PropertyDescriptorMap {
-    const to: PropertyDescriptorMap = { __proto__: null } as any;
-    // The following can be ignored because for now, redOwnKeys is
-    // intentionally empty. In getRedReferences, see the following:
-    //
-    //      record.WindowPropertiesProtoOwnKeys = [];
-    //
-    // istanbul ignore next
-    for (let i = 0, len = redOwnKeys.length; i < len; i += 1) {
-        const key = redOwnKeys[i] as string;
-        to[key] = blueDescriptors[key];
-    }
-    return to;
-}
-
 export function tameDOM(
     env: VirtualEnvironment,
     blueRefs: CachedBlueReferencesRecord,
-    redRefs: RedReferencesRecord,
     endowmentsDescriptors: PropertyDescriptorMap
 ) {
     // adjusting proto chain of window.document
     env.remapProto(blueRefs.document, blueRefs.DocumentProto);
-    const globalDescriptors = aggregateWindowDescriptors(
-        redRefs.windowOwnKeys,
-        blueRefs.windowDescriptors,
-        endowmentsDescriptors
-    );
-    const WindowPropertiesDescriptors = aggregateWindowPropertiesDescriptors(
-        redRefs.WindowPropertiesProtoOwnKeys,
-        blueRefs.WindowPropertiesProtoDescriptors
-    );
+    const globalDescriptors = filterWindowDescriptors(endowmentsDescriptors);
     // remapping globals
     env.remap(blueRefs.window, globalDescriptors);
     // remapping unforgeable objects
     env.remap(blueRefs.EventTargetProto, blueRefs.EventTargetProtoDescriptors);
-    env.remap(blueRefs.WindowPropertiesProto, WindowPropertiesDescriptors);
     env.remap(blueRefs.WindowProto, blueRefs.WindowProtoDescriptors);
+    /**
```

Comment:
again, removing more unnecessary initialization steps, just commenting why we are skipping it.